core/primitives/src/runtime/config_store.rs

```diff
@@ -0,0 +1,147 @@
+use crate::runtime::config::RuntimeConfig;
+use crate::types::ProtocolVersion;
+use std::collections::BTreeMap;
+use std::iter::FromIterator;
+use std::ops::Bound;
+use std::sync::Arc;
+
+macro_rules! include_config {
+    ($file:expr) => {
+        include_bytes!(concat!("../../../../nearcore/res/runtime_configs/", $file))
+    };
+}
+
+/// Stores pairs of protocol versions for which runtime config was updated and
+/// the new runtime config in bytes.
+/// Protocol versions are given in increasing order. First one is always 0, so that each version is
+/// mapped to some config.
+static CONFIGS: [(ProtocolVersion, &[u8]); 2] =
+    [(0, include_config!("29.json")), (42, include_config!("42.json"))];
+
+/// Stores runtime config for each protocol version where it was updated.
+#[derive(Debug)]
+pub struct RuntimeConfigStore {
+    /// Maps protocol version to the config.
+    store: BTreeMap<ProtocolVersion, Arc<RuntimeConfig>>,
+}
+
+impl RuntimeConfigStore {
+    /// Constructs a new store.
+    ///
+    /// If min_allowed_top_level_account_length is Some, overrides the value in
+    /// account_creation_config in all runtime configs in resulting store.
+    /// TODO #4775: disable this override for future protocol versions, because it is needed only
+    /// for old testnet versions.
+    pub fn new(min_allowed_top_level_account_length: Option<u8>) -> Self {
```

Comment:
I am not sure whether this is the best approach. I remember that on betanet we also have registrar account id set to something different from mainnet.

I think we can always load the latest runtime config to override the existing one based on protocol version here. A more general approach could be to expand this to a "protocol config" and override what is currently `GenesisConfig` (most of it) based on the protocol version, but that likely requires more refactoring and could be done later.

Reply:
Yeah, plus one to the idea that, if you pass one config parameter, you might instead allow the caller to do free-form modifications. That way, the callee isn't coupled to the specifics of what needs to be overridden. 

I usually write such API as:

```rust
impl RuntimeConfigStore {
    fn override_genesis_config(self, f: impl FnOnce(&mut RuntimeConfig)) -> RuntimeConfigStore
}
```